esp-wifi/src/wifi/mod.rs

```diff
@@ -1303,6 +1306,8 @@ pub(crate) mod embassy {
             cx: &mut core::task::Context,
         ) -> Option<(Self::RxToken<'_>, Self::TxToken<'_>)> {
             RECEIVE_WAKER.register(cx.waker());
+            TRANSMIT_WAKER.register(cx.waker());
```

Comment:
Why do we register the TRANSMIT_WAKER here?

Reply:
It's technically possible that we return None because we can't create a TxToken. TRANSMIT_WAKER is supposed to signal a change in this state. I don't think this is necessary, as at this point the waker is probably registered to the correct task and that can't change in embassy. Since we are registering wakers regardless of whether they are actually needed, I was thinking "why not".